index.html

```diff
@@ -259,10 +259,200 @@ <h1>Architecture</h1>
+			To clarify, if the D-Client only supports one Introduction mechanism and that mechanism
+			can only produce one URL, then the "merge" is trivial.
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-td-identify">A D-Client MUST be able to identify whether a TD fetched from an 
+			Introduction URL is a ThingDirectory or a ThingLink.</span>
+			This does not imply that the D-Client needs to take any action on such TDs; it just needs to be able
+			to check the <code>@type</code> field and make this distinction.
+                        A D-Client can decide whether or not to follow links or fetch TDD contents.
+	                There are some use cases where a Consumer may not expand certain URLs, for example links pointing
+	                at external resources, or when a TDD contains many TDs and fetching them all
+	                would exceed the Consumer's memory processing capabilities.
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-fetch-tdd">A D-Client MAY fetch additional TDs from any Exploration mechanism
+			described in its initial set of TDs (including, in particular, Thing Description Directories) 
+			and add them into the set of TD results.</span>
+			This only adds the results of fetching TDs from a Exploration mechanism to the set of
+			results. These new results do not delete the original TD describing the Exploration mechanism. 
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-fetch-links">A D-Client MAY fetch source TDs from the targets of the links in a Thing Link
+			described in its initial set of TDs and add them into the set of TD results.</span>
+			This only adds the results of fetching TDs from a Thing Link to the set of
+			results. These new results do not delete the original TD describing the Exploration mechanism. 
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-fetch-iteration">A D-Client MAY fetch additional TDs iteratively from any Thing Link or
+			Exploration mechanism described in its set of TDs and add them into the set of TD results.</span>
+			This only adds the results of fetching TDs from a Thing Link to the set of
+			results. These new results do not delete the original TD describing the Exploration mechanism. 
+		</li>
+		<li><span class="rfc2119-assertion" id="consumer-termination">A D-Client MAY terminate fetching additional TDs at any point or for
+			any reason.</span>
+	        </li>
+		<li><span class="rfc2119-assertion" id="consumer-any-order">A D-Client MAY fetch additional TDs by following links or fetching
+			additional TDs from Exploration mechanisms (e.g. TDDs) in any order.</span>
+			The D-Client can fetch additional TDs in any order from any Exploration mechanism and is not
+			required to use all of them.
+	        </li>
+		<li><span class="rfc2119-assertion" id="consumer-track">A D-Client MUST track (by id) which TDs describing links or Exploration mechanisms
```

Comment:
Is `id` mandatory for ThingLinks and ThingDirectories? cause, if my mind is not failing me, it is not for generic TDs. For TMs loop cycle detection we are using the `href` of links as an ID.

Reply:
Ack, good point.  A Thing Link or a Thing Directory without an id would be a pain.   I guess we could add "or href, if no id is available".

Reply:
I changed this to just say that loops should be detected, not how.  Could also use full-result hashes, but I think this is an implementation decision, not a requirement.